hazelcast/src/test/java/com/hazelcast/sql/impl/expression/datetime/FormatterTest.java

```diff
@@ -0,0 +1,258 @@
+
+        f = new Formatter("\"Quarter\" YYYY-\"Q\"Q is in the CCth century.");
+        check(YearMonth.of(2022, 9), f, "Quarter 2022-Q3 is in the 21st century.");
+
+        f = new Formatter("Roman numeral for Month is RM.");
+        check(YearMonth.of(2022, 9), f, "Roman numeral for September is IX.");
+
+        f = new Formatter("Plato founded the \"Academy\" in c. YYYY AD.");
+        check(Year.of(-386), f, "Plato founded the Academy in c. 387 BC.");
+
+        f = new Formatter("Plato \"Akademi\"'yi TMA.D. YYYY civarında kurdu.");
+        check(Year.of(-386), f, "Plato Akademi'yi M.Ö. 387 civarında kurdu.", "tr-TR");
+
+        f = new Formatter("AD(=BC) A.D.(=B.C.) ad(=bc) a.d.(=b.c.)");
+        check(Year.of(0), f, "BC(=BC) B.C.(=B.C.) bc(=bc) b.c.(=b.c.)");
+
+        f = new Formatter("The Halley's closest approach to the Earth was on DD Month YYYY AD (the Jth \"Julian day\").");
+        check(LocalDate.of(837, 2, 28), f,
+                "The Halley's closest approach to the Earth was on 28 February 837 AD (the 2026827th Julian day).");
+    }
+
+    @Test
+    public void testWeekDates() {
+        Formatter f = new Formatter("IYYY-\"W\"FMIW-ID");
+        check(LocalDate.of(2022, 10, 25), f, "2022-W43-2");
+        check(LocalDate.of(2019, 12, 30), f, "2020-W01-1");
+
+        f = new Formatter("YYYY-FMMM-FMDD is the IDDDth \"day\" of week-year IYYY.");
+        check(LocalDate.of(2008, 12, 29), f, "2008-12-29 is the 1st day of week-year 2009.");
+        check(LocalDate.of(2010, 1, 3), f, "2010-01-03 is the 371st day of week-year 2009.");
+
+        f = new Formatter("FMIYYY(IYYY) FMIYY(IYY) FMIY(IY) FMI(I)");
+        check(LocalDate.of(1, 1, 1), f, "0001(1) 001(1) 01(1) 1(1)");
+    }
+
+    @Test
+    public void testRounding() {
+        Formatter f = new Formatter("FM0.9");
+        check(0.15, f, "0.2");
+    }
+
+    @Test
+    public void testGrouping() {
+        Formatter f = new Formatter("9G99G999D990");
+        check(-123456.78, f, "-1,23,456.780");
+    }
+
+    @Test
+    public void testSign() {
+        Formatter f = new Formatter("0.9S");
+        check( 1, f, "1+  ");
+        check( 0, f, "0+  ");
+        check(-1, f, "1-  ");
+
+        check(4.5,  "  4.5 ",  "< 4.5>",  "99.9BR", "BR99.9", "BR99.9BR", "BR99.9B");
+        check(4.5,  "  4.5 ",  " <4.5>",  "99.9B",   "B99.9",  "B99.9B");
+        check(4.5,  " +4.5 ",  " <4.5>",  "P9.9B",   "BP9.9",  "PB9.9");
+
+        check(4.5,  "+ 4.5",  "- 4.5",  "SG99.9");
+        check(4.5,  " 4.5+",  " 4.5-",  "99.9SG", "99.9S");
+        check(4.5,  " +4.5",  " -4.5",  "S99.9");
+
+        check(4.5,  "  4.5",  "- 4.5",  "MI99.9");
+        check(4.5,  " 4.5 ",  " 4.5-",  "99.9MI", "99.9M");
+        check(4.5,  "  4.5",  " -4.5",  "M99.9");
+
+        check(4.5,  "+ 4.5",  "  4.5",  "PL99.9");
+        check(4.5,  " 4.5+",  " 4.5 ",  "99.9PL", "99.9P");
+        check(4.5,  " +4.5",  "  4.5",  "P99.9");
```

Comment:
This check fails:
```
check(0,  "+ 0",  "+ 0",  "PL9");
```
```
org.junit.ComparisonFailure: 
Expected :+ 0
Actual   :+ 
```

Reply:
`"PL9"` doesn't introduce an extra space between sign and digit. This will work: `check(0,  "+0",  " 0",  "PL0");`

Reply:
This is what PG returns. If there's only PL pattern, the number still will contain `-`, for which the space is reserved. In our case, `to_char(-1, 'PL0')` returns `"1"`, which is wrong in 2 ways:
- the number is actually incorrect and should rather be `"#"`, if can't contain any minus sign.
- there should be a space reserved for the `+` sign.

Reply:
1. > If no explicit provision is made for a sign in `to_char()`'s pattern, one column will be reserved for the sign, and it will be anchored to (appear just left of) the number.

   Our implementation complies with Postgres' documented behavior and since you specify a `PL` pattern, no column is reserved for the minus sign.
2. The number is not incorrect. When checking against overflow, sign is not taken into account because it is a representation issue. Think about putting brackets instead of a minus sign; it is unconventional (for programmers), but correct. You can choose to omit minus sign and print only plus sign, i.e. reverse our daily convention; the interpretation is up to you.
3. There is already a space reserved for the plus sign and it is occupied if the number is positive.

Reply:
The general rule is that it should never produce incorrect number. It can be imprecise (rounded), but not incorrect. A negative number without any sigh _is_ incorrect. Brackets are a common convention in finance, but using `+` sign for positive and no sign for negative values is not.

We should follow the PG behavior.